dashboard/app/models/sections/omni_auth_section.rb

```diff
@@ -39,7 +39,15 @@ def self.from_omniauth(code:, type:, owner_id:, students:, section_name: I18n.t(
     oauth_section = with_deleted.where(code: code).first_or_create
 
     oauth_section.name = section_name || I18n.t('sections.default_name', default: 'Untitled Section')
-    oauth_section.user_id = owner_id
+
+    # Add the user as an owner if the section does not exist. Otherwise add as a coteacher to existing section.
+    if oauth_section.user_id.nil? || oauth_section.deleted?
+      oauth_section.user_id = owner_id
+    else
+      # create a section instructor record if one doesn't exist
+      section_instructor = oauth_section.section_instructors.where(instructor_id: owner_id).first_or_create(status: :active)
```

Comment:
I suspect you'll need this too, and a test would be great to make sure that's all.

```suggestion
      section_instructor = oauth_section.section_instructors.with_deleted.where(instructor_id: owner_id).first_or_create(status: :active)
```

Reply:
Also `first_or_create` only passes the status argument if a new record is created, it won't update the status on an existing record.

Reply:
Added